.github/workflows/pr-intake.yml

```diff
@@ -0,0 +1,157 @@
+name: pr-intake
+
+on:
+    pull_request:
+        branches: 
+        - main
+        - 'releases/**'     
+        paths:
+        - 'Examples/**'
+        - 'SemiconductorTestLibrary.Abstractions/**'
+        - 'SemiconductorTestLibrary.Extensions/**'
+        - 'SemiconductorTestLibrary.TestStandSteps/**'
+        - 'TestAssets/**'
+        - '.github/**'
+
+env:
+    GitHub_Repo_Name: "semi-test-library-dotnet"
+    PR_Number: "#${{ github.event.number }}"
+
+jobs:
+  wait-for-pr-build-and-test-workflows:
+    runs-on: ubuntu-latest
+    environment: gh-action-testing
+    steps:
+      # Before fetching the run ID, we need to wait for the PR Build and PR Test Workflow run to start as it uses in_progress type of workflow_run event.
+      - name: Wait for PR Build and PR Test Workflow to Start
+        run: sleep 10
+
+      - name: Get PR Build Workflow Run ID
+        id: get_build_runid
+        run: |
+          WORKFLOW_NAME="pr-build"
+
+          WORKFLOW_ID=$(curl -s "https://api.github.com/repos/ni/${{ env.GitHub_Repo_Name }}/actions/workflows" \
+                             | jq -r ".workflows[] | select(.name == \"$WORKFLOW_NAME\") | .id")
+
+          # Fetch runs of Workflow
+          response=$(curl -s -w "%{http_code}" "https://api.github.com/repos/ni/${{ env.GitHub_Repo_Name }}/actions/workflows/$WORKFLOW_ID/runs" \
+                   -o runs.json)
+
+          if [ $response -ne 200 ]; then
+            echo "Failed to fetch runs of Workflow - $WORKFLOW_NAME. Exiting."
+            exit 1
+          fi
+
+          # Assuming that the PR Build Workflow run has a display title containing the PR number.
+          run_id=$(jq -r '.workflow_runs[] | select(.display_title | contains(${{ env.PR_Number }})) | .id' runs.json)
+          
+          echo "Fetched build run_id: $run_id"
+          echo "::set-output name=build_run_id::$run_id"
+
+      - name: Get PR Test Workflow Run ID
+        id: get_test_runid
+        run: |
+          WORKFLOW_NAME="pr-test"
+
+          WORKFLOW_ID=$(curl -s "https://api.github.com/repos/ni/${{ env.GitHub_Repo_Name }}/actions/workflows" \
+                              | jq -r ".workflows[] | select(.name == \"$WORKFLOW_NAME\") | .id")
+
+          # Fetch runs of Workflow
+          response=$(curl -s -w "%{http_code}" "https://api.github.com/repos/ni/${{ env.GitHub_Repo_Name }}/actions/workflows/$WORKFLOW_ID/runs" \
```

Comment:
@Tanish-NI, shouldn't the valid response check also be done in the other run steps as well, just to be safe: `Wait for Specific Build Workflow Run to Complete`, `Wait for Specific Test Workflow Run to Complete`, `Check Conclusion of Build Workflow`, and `Check Conclusion of Test Workflow`?

Reply:
Sure @Mattjet27 !
Added the HTTP status check after each API call.